lib/key/forwarding.js

```diff
@@ -0,0 +1,86 @@
+import BigIntegerInterface from '@openpgp/noble-hashes/esm/biginteger/interface';
+import { KDFParams } from '../openpgp';
+import { generateKey, reformatKey } from './utils';
+
+let loadedBigInteger = false;
+const getBigInteger = async () => {
+    // Temporary function to be dropped once openpgpjs v6 (which will bundle noble-hashes) is integrated.
+    // openpgpjs v5 internally includes a BigInteger implementation, but it is not exported.
+    // noble-hashes's BigInteger export automatically imports BN.js (as BigInt fallback),
+    // instead we only import it if needed to minimise the bundle size.
+    if (loadedBigInteger) return BigIntegerInterface;
+
+    const detectBigInt = () => typeof BigInt !== 'undefined';
+    if (detectBigInt()) {
+        // NativeBigInteger is small, so it could also be imported always
+        const { default: NativeBigInteger } = await import('@openpgp/noble-hashes/esm/biginteger/native.interface');
+        BigIntegerInterface.setImplementation(NativeBigInteger);
+    } else {
+        const { default: FallbackBigInteger } = await import('@openpgp/noble-hashes/esm/biginteger/bn.interface');
+        BigIntegerInterface.setImplementation(FallbackBigInteger);
+    }
+    loadedBigInteger = true;
+
+    return BigIntegerInterface;
+};
+
+export async function computeProxyParameter(originalSecret, finalRecipientSecret) {
+    const BigInteger = await getBigInteger();
+
+    const dB = BigInteger.new(originalSecret);
+    const dC = BigInteger.new(finalRecipientSecret);
+    const n = BigInteger.new('0x1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed'); // 2^252 + 0x14def9dea2f79cd65812631a5cf5d3ed
+    const proxyParameter = dC.modInv(n).mul(dB).mod(n).toUint8Array('le');
+
+    return proxyParameter;
+}
+
+/**
+ * Generate a forwarding key for the final recipient, as well as the corresponding proxy factor.
+ * The key in input must be a v4 primary key and must have at least one ECDH subkey using curve25519 (legacy format)
+ * @param originalKey       ECC primary key of original recipient
+ * @param forwardingUserIds array of user IDs of forwarding key
```

Comment:
`forwarding` is too vague IMO, it should be either `forwardee` or `finalRecipient` (I would go with the former as that's the term used in the draft spec)

Reply:
I think Aron wanted to move away from 'forwardee' and similar since it's very confusing to distinguish what is what.
It also shouldn't be 'finalRecipient' since effectively we want this to be the original recipient userID.
The field description makes it clearer what the value is, but maybe I can extend the name to `userIDsForForwardingKey` ?